src/backend/linux_raw/fs/syscalls.rs

```diff
@@ -1458,6 +1458,45 @@ pub(crate) fn copy_file_range(
     }
 }
 
+const NAME_MAX: usize = 255;
+const SHM_DIR: &[u8] = b"/dev/shm/";
+
+fn get_shm_name(name: &CStr) -> io::Result<[u8; NAME_MAX + SHM_DIR.len() + 1]> {
+    let name = name.to_bytes();
+
+    if name.len() > NAME_MAX {
+        return Err(io::Errno::NAMETOOLONG);
+    }
+
+    let num_slashes = name.into_iter().take_while(|x| **x == b'/').count();
+    let after_slashes = &name[num_slashes..];
+    if after_slashes.is_empty()
+        || after_slashes == b"."
+        || after_slashes == b".."
+        || after_slashes.contains(&b'/')
+    {
+        return Err(io::Errno::INVAL);
+    }
+
+    let mut path = [0; NAME_MAX + SHM_DIR.len() + 1];
+    path[..SHM_DIR.len()].copy_from_slice(SHM_DIR);
+    path[SHM_DIR.len()..SHM_DIR.len() + name.len()].copy_from_slice(name);
+    Ok(path)
+}
+
+pub(crate) fn shm_open(name: &CStr, oflags: OFlags, mode: Mode) -> io::Result<OwnedFd> {
+    let path = get_shm_name(name)?;
+    // In Rust 1.69, can use `from_bytes_until_nul`
+    let path = &path[..=path.iter().position(|x| *x == 0).unwrap()];
+    open(CStr::from_bytes_with_nul(&path).unwrap(), oflags, mode)
+}
+
+pub(crate) fn shm_unlink(name: &CStr) -> io::Result<()> {
+    let path = get_shm_name(name)?;
+    let path = &path[..=path.iter().position(|x| *x == 0).unwrap()];
```

Comment:
We could avoid these `position` calls by having `get_shm_name` also return the length of the name, which it already knows.

Reply:
Good point. I originally wrote it with `from_bytes_until_nul`, but then CI complained since that's too new of a standard library addition.

Either `from_bytes_with_nul` or `from_bytes_until_nul` still need to traverse the string. `method.from_bytes_with_nul_unchecked` would avoid that.